src/plaquette/codes/__init__.py

```diff
@@ -7,13 +7,41 @@
 from enum import IntEnum
 from functools import cached_property
 
+import igraph as ig
 import numpy as np
 
 from plaquette import graph, pauli
 from plaquette.codes.utils import generate_factorized_checks
 from plaquette.pauli import Tableau, argsort_operators_ref
 
 
+def generate_layout_coordinates(
+    nodes: list[int], edges: list[tuple[int, int]]
+) -> dict[int, graph.Position]:
+    """Generate the layout coordinates for the tanner and orchestration graphs.
+
+    Args:
+        nodes: The list of node indices.
+        edges: The list of edge tuples of the graph.
+
+    Returns:
+        A dictionary of node positions with keys as node indices.
+    """
+    g = ig.Graph(vertex_attrs={"label": nodes}, edges=edges)
+    layout = g.layout_auto()
+
+    # Convert the layout to a dictionary format
+    coordinates = {}
+    for idx, node in enumerate(nodes):
+        coord = layout[idx]
+        if len(coord) == 2:
+            coordinates[node] = graph.Position(x=coord[0], y=coord[1], z=0)
+        else:
+            coordinates[node] = graph.Position(x=coord[0], y=coord[1], z=coord[2])
```

Comment:
Minor: a potential small refactor here (note: not tested).
```suggestion
        z_coord = 0 if len(coord) == 2 else coord[2]
        coordinates[node] = graph.Position(x=coord[0], y=coord[1], z=z_coord)
```